enhancements/node/lazy-image-pull.md

```diff
@@ -0,0 +1,283 @@
+* As a developer, I want to deploy applications packaged in large container images while minimizing the impact of image size on pod startup time.
+
+### Goals
+
+* Package upstream [Stargz Store plugin](https://github.com/containerd/stargz-snapshotter) for the installation on RHCOS
+* Enable lazy image pulling in Openshift using [Stargz Store plugin](https://github.com/containerd/stargz-snapshotter) with [zstd:chunked](https://github.com/containers/storage/pull/775) image format.
+* Support lazy image pulling only on the worker nodes (or subset of worker nodes).
+* Add documenatation to easily create `zstd:chunked` images from scratch as well as converting existing images to `zstd:chunked` format.
+* Add support for drop-in config in [container storage config](https://github.com/containers/storage/blob/main/pkg/config/config.go)
+
+### Non-Goals
+
+* Enabling lazy image pulling on the control plane nodes.
+* Support for lazy image pull format other than `zstd:chunked`
+
+## Proposal
+
+In order to lazy pull images, we need to have [Stargz Store plugin](https://github.com/containerd/stargz-snapshotter/blob/aaa46a75dd97e401025f82630c9d3d4e41c9f670/docs/INSTALL.md#install-stargz-store-for-cri-opodman-with-systemd) and [its authentication helper](https://github.com/containers/image/pull/2417) installed on the node as well as `additionallayerstores` configuration for container storage library. [ContainerRuntimeConfig](https://github.com/openshift/machine-config-operator/blob/release-4.16/docs/ContainerRuntimeConfigDesign.md) will be used to deploy the required configuration on the target nodes.
+
+
+### Workflow Description
+
+* A cluster administrator will create or edit the [ContainerRuntimeConfig](https://github.com/openshift/machine-config-operator/blob/release-4.16/docs/ContainerRuntimeConfigDesign.md) object to enable lazy image pulling support by,
+```yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: ContainerRuntimeConfig
+metadata:
+ name: lazy-image-pull
+spec:
+ machineConfigPoolSelector:
+   matchLabels:
+     pools.operator.machineconfiguration.openshift.io/worker: ""
+ containerRuntimeConfig:
+     lazyImagePull:
+        enabled: true
+```
+### API Extensions
+
+Extend existing [ContainerRuntimeConfiguration](https://github.com/openshift/api/blob/ff84c2c732279b16baccf08c7dfc9ff8719c4807/machineconfiguration/v1/types.go#L749C6-L749C35) to hold the required configuration for lazy image pulling.
+
+```golang
+// ContainerRuntimeConfiguration defines the tuneables of the container storage
+type ContainerRuntimeConfiguration struct {
+   // +kubebuilder:validation:Optional
+   LazyImagePullConfig *LazyImagePullConfiguration `json:"lazyImagePullConfig, omitempty"`
+}
+```
+
+```golang
+type LazyImagePullConfiguration struct {
+   // +kubebuilder:validation:Required
+   Enable bool `json:"enable"`
```

Comment:
FYI openshift api doesn't anymore allow booleans because they're not extendable, you'll need to add an enum

Reply:
Thanks for the review. Fixed to avoid using booleans.